app/src/main/java/com/fcaputo/parkingapp/mvp/view/ReservationView.kt

```diff
@@ -0,0 +1,157 @@
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.google.android.material.datepicker.CalendarConstraints
+import com.google.android.material.datepicker.DateValidatorPointForward
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.timepicker.MaterialTimePicker
+import com.google.android.material.timepicker.TimeFormat
+import java.time.LocalTime
+import java.util.Date
+
+class ReservationView(activity: Activity) : ActivityView(activity), ReservationContract.View {
+    private var binding: ActivityReservationBinding = ActivityReservationBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun showParkingLotSize(size: Int) {
+        val sizeString = activity?.getString(R.string.reservation_parking_lot_size_helper, size)
+        binding.tilParkingLot.helperText = sizeString
+    }
+
+    override fun onSaveButton(onClick: () -> Unit) {
+        binding.submitButton.setOnClickListener { onClick() }
+    }
+
+    override fun onDateInputPressed(onInputClick: (CharSequence?, Int) -> Unit) {
+        for (tInput in listOf(binding.etStartDate, binding.etEndDate)) {
+            tInput.setOnClickListener {
+                onInputClick((it as TextInputEditText).text, it.id)
+            }
+        }
+    }
+
+    override fun showDatePicker(dateSelected: Date, editTextId: Int) {
+        val calendarConstraint = CalendarConstraints.Builder()
+            .setValidator(DateValidatorPointForward.now())
+            .build()
+
+        val datePicker = MaterialDatePicker.Builder.datePicker()
+            .setCalendarConstraints(calendarConstraint)
+            .setSelection(dateSelected.time)
+            .build()
+
+        datePicker.addOnPositiveButtonClickListener {
+            val dateString = DateTimeCustomFormat.formatDate(Date(it))
+            val editText = ViewBindings.findChildViewById<TextInputEditText>(binding.root, editTextId)
+            editText?.setText(dateString)
+        }
+
+        datePicker.show((activity as AppCompatActivity).supportFragmentManager, activity?.getString(R.string.date_picker_tag))
+    }
+
+    override fun onTimeInputPressed(onInputClick: (CharSequence?, Int) -> Unit) {
+        for (tInput in listOf(binding.etStartTime, binding.etEndTime)) {
+            tInput.setOnClickListener {
+                onInputClick((it as TextInputEditText).text, it.id)
+            }
+        }
+    }
+
+    override fun showTimePicker(timeSelected: LocalTime, editTextId: Int) {
+        val timePicker = MaterialTimePicker.Builder()
+            .setTimeFormat(TimeFormat.CLOCK_12H)
+            .setHour(timeSelected.hour)
+            .setMinute(timeSelected.minute)
+            //.setTitleText("Select reservation time")
```

Comment:
please, uncomment or remove this commented code

Reply:
Done

Reply:
Done